widgets-tileset/index.ts

```diff
@@ -0,0 +1,356 @@
+
+  const percent = fraction_dropped_per_zoom[clampedZoom]
+  return percent
+}
+
+function clamp(n: number, min: number, max: number) {
+  return Math.min(Math.max(n, min), max);
+}
+
+
+async function initSource() {
+  dataSource = await vectorTilesetSource({
+    ...cartoConfig,
+    tableName: 'cartodb-on-gcp-pm-team.amanzanares_opensource_demo.national_water_model_tileset_final_test_4',
+  })
+  return dataSource;
+}
+
+// SPATIAL FILTER
+// prepare a function to get the new viewport state, that we'll pass debounced to our widgets to minimize requests
+
+let currentZoom = INITIAL_VIEW_STATE.zoom;
+let viewportSpatialFilter;
+
+const debouncedUpdateSpatialFilter = debounce(viewState => {
+  currentZoom = viewState.zoom;
+  const viewport = new WebMercatorViewport(viewState);
+  viewportSpatialFilter = createViewportSpatialFilter(viewport.getBounds());
+  renderWidgets();
+}, 300);
+
+// sync deckgl map after user interaction, obtain new viewport after
+
+deck.setProps({
+  onViewStateChange: ({viewState}) => {
+    const {longitude, latitude, ...rest} = viewState;
+    map.jumpTo({center: [longitude, latitude], ...rest});
+    debouncedUpdateSpatialFilter(viewState);
+  }
+});
+
+// COLUMN FILTERS
+// prepare filters
+
+let filters = {};
+
+// filtering from the widget, including a clear filters button
+
+const clearFiltersButton = document.querySelector<HTMLElement>('#clearHistogramFilter');
+
+clearFiltersButton?.addEventListener('click', () => {
+  clearHistogramFilter();
+});
+
+function filterViaHistogram(dataIndex) {
+  clearFiltersButton.style.display = 'inherit';
+
+  removeFilter(filters, {
+    column: 'streamOrder'
+  });
+
+  const minValue = histogramTicks[dataIndex];
+  const maxValue = histogramTicks[dataIndex + 1] - 0.0001;
+
+  if (dataIndex === histogramTicks.length - 1) {
+    // For the last category (> 600), use CLOSED_OPEN
+    addFilter(filters, {
+      column: 'streamOrder',
+      type: FilterType.CLOSED_OPEN,
+      values: [[minValue, Infinity]]
+    });
+  } else {
+    // For first and middle categories, use BETWEEN
+    addFilter(filters, {
+      column: 'streamOrder',
+      type: FilterType.BETWEEN,
+      values: [[minValue, maxValue]]
+    });
+  }
+
+  initialize();
+}
+
+function clearHistogramFilter() {
+  clearFiltersButton.style.display = 'none';
+
+  removeFilter(filters, {
+    column: 'income_per_capita'
+  });
+
+  initialize();
+}
+
+// RENDER
+// prepare ticks for our widget
+
+const histogramTicks = [1, 2, 3, 4, 5, 6, 7, 8, 9, 10];
+// Array.from({length: 31}, (_, i) => i * (600 / 30))
+// render Widgets function
+
+async function renderWidgets() {
+  // Exit if dataSource is not ready
+  if (!dataSource) {
+    return;
+  }
+
+  // Exit if tiles are not loaded by layer viewport
+  if (!tilesLoaded) {
+    return;
+  }
+
+  const droppingPercent = getDroppingPercent(dataSource, currentZoom);
+  droppingWarningSmall.classList.add('hidden');
+  droppingWarningBig.classList.add('hidden');
```

Comment:
Could we instead show/hide these messages with...

```js
el.style.display = 'none';
el.style.display = '';
```

Just in the interest of showing best practice , ideally they'd be shown/hidden from accessibility tools like screenreaders as well as visually.

Reply:
I built this way to have the elements initially hidden before the JS loads.
I could use this other behavior and add the initial `display: none` to the widgets class